cura/CuraApplication.py

```diff
@@ -1600,8 +1600,11 @@ def log(self, msg):
 
     openProjectFile = pyqtSignal(QUrl, arguments = ["project_file"])  # Emitted when a project file is about to open.
 
-    @pyqtSlot(QUrl, bool)
-    def readLocalFile(self, file, skip_project_file_check = False):
+    @pyqtSlot(QUrl, str)
+    ## Open a local file
+    # \param project_mode How to handle project files. Either None(default): Follow user preference, "open_as_model" or
+    # "open_as_project". This parameter is only considered if the file is a project file.
+    def readLocalFile(self, file: QUrl, project_mode: Optional[str] = None):
```

Comment:
This function is used in other places (such as recentFilesMenu, line 66). There it's still giving a bool instead of a string.